### ISSUE TITLE
Canvas Instruments

### ISSUE THREAD

**Issue by D-ECHO** [CRJ700-family-team/CRJ700-family#2]
work done by @d-jsb, extract/incorporate framework into FGAddon, then bring the instruments to this repository
https://forum.flightgear.org/viewtopic.php?f=4&t=31088&p=359105#p359093

**Comment by DanielMeissner** [comment]
Just to give you a hint of what's to come (I hope soon?) :P Henning really did great work on this!
![fgfs-20200102154657](https://user-images.githubusercontent.com/44880341/71676131-b2146900-2d7f-11ea-9cdb-0784fb421095.png)

By the way: Working Nav source selection and finally correct course pointer colors :-)

**Comment by D-ECHO** [comment]
Really cool! Definitely looking forward to that

**Comment by tdammers** [comment]
> By the way: Working Nav source selection and finally correct course pointer colors :-)

Oooooh... this is something I've wanted for a long time, but never got around to fixing it...

**Comment by tdammers** [comment]
Quick question here: where does the code for that live? I tried the `canvas-efis-beta` branch on @DanielMeissner's fork, ~but that crashes FG during startup~ Never mind, I was using a broken attempted merge instead of the real thing.

**Comment by tdammers** [comment]
OK, impressions so far:

- Performance is rather bad. I usually get around 10-15 fps, but with the canvas version, frame rates drop to 5-8, occasionally down to 3, iow., practically unflyable. This is something I have observed with pretty much all canvas cockpits, and it is one of the reasons why I fly the CRJ so much. I'm not sure whether it can be fixed completely, but I believe reducing canvas update rates, and reusing the captain's screens for the FO (thus reducing the number of canvas screens to be rendered) can help mitigate the problem somewhat. All of those should be implemented as optional configuration settings, obviously.
- The MFD doesn't look anything like the CRJ one, looks more like a Boeing one. In the CRJ, the compass / sector should only occupy the lower part of the MFD, while the upper part lists upcoming waypoints. See for example here: http://www.smartcockpit.com/docs/CRJ-00_and_00-Flight_Instrumenst.pdf (page 12 has a useful figure). Then again, the one in the screenshot does look about right, so I'm pretty sure I haven't been looking at the latest version (in which case, where is it?)
- It does look pretty.

**Comment by D-ECHO** [comment]
@tdammers canvas-efis-beta is _not_ the correct version, it's afaik on d-jsb's computer (only?)

I think we should delay this discussion until it's released. @d-jsb please (really) don't feel pressured, take your time :)

**Comment by tdammers** [comment]
Right.

FWIW, I'd much prefer sampling a WIP version over waiting for this to evolve secretly. I don't see what harm there could be in sharing what's currently there.

**Comment by D-ECHO** [comment]
I don't think this is about keeping something secret, it's just that everyone has real-life obligues to do.

**Comment by tdammers** [comment]
Of course. So do I.

On Fri, Jan 03, 2020 at 01:33:51AM -0800, D-ECHO wrote:
> I don't think this is about keeping something secret, it's just that everyone has real-life obligues to do.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/CRJ700-family-team/CRJ700-family/issues/2#issuecomment-570517408

-- 
Tobias Dammers - tdammers@gmail.com

**Comment by D-ECHO** [comment]
Do the instruments already ported to canvas also inclue the frequency selectors (RMU?) on the rear pedestal?

**Comment by DanielMeissner** [comment]
Not as far as I know...

**Comment by D-ECHO** [comment]
Okay thx

When the FDM and cockpit 3d is in a good shape, I could add them

**Comment by d-jsb** [comment]
It is finally online. I just pushed one big commit to my https://github.com/d-jsb/CRJ700-family master branch.
I guess it contains the work of the last two or three years. 
While I am really happy to have this finally uploaded, I am sure there will be a lot of small and not so small issues. The most obvious being not updated documentation.

**Comment by D-ECHO** [comment]
Great work! I'm currently cloning your repo into another folder and will post some feedback once done.

**Comment by D-ECHO** [comment]
On first test it seems very nice, a lot of functionality and I haven't found any bugs yet :) Thank you!

**Comment by d-jsb** [comment]
I recommend running FG with console window - do not worry about the "ERROR: Cannot add listener to tied property <blabla>" none of them caused a real problem so far - I reported this to the dev mailing list already, it is only an issue under certain conditions which does not seem to apply to our use case.
You may want to have a look to checklists and tutorials (help menu), the "Bombardier CRJ700" menu, the items in the "Ground ops" menu etc.
Hint: if you want to use in-flight start from the menu on ground, open the config dialog from the same menu and enable Developer mode (aircraft up and running in under a second ;))

**Comment by d-jsb** [comment]
oh, and I guess you need to enable horizontal stab trim from the config dialog or AP may do weird things - old FG versions (yasim version to be accurate) won't support it. The point is that you can trim the stabilizer since yasim version 2018.1 (this is how real airliners do it) so you do not have to rely on elevator trim only (elevator trim in real live is mostly seen on small GA aircraft, not airliners). This leads to more elevator authority available on approach.

**Comment by tdammers** [comment]
I am pretty sure I had already fixed the elevator / stab trim thing - on
recent yasim versions, yoke and AP control elevator, and AP trim and manual
trim control stabilizer; on older yasim, stab trim is emulated with
elevator trim, which is as good an approximation as you will get. There was
also some reverse logic to make the yasim solver work for both cases.

I can't remember whether I removed the stab trim config option, but I
reckon with the yasim version compat stuff in place, it would be
unnecessary.

**Comment by d-jsb** [comment]
Ok, I did not check the history of this repo yet. I guess there are some improvements from you all that have to be merged. I guess I can start this tomorrow or at the weekend...
